### PR TITLE
Prevent infinite loop in `Command::Base#step` & add configurable wait time

### PR DESCRIPTION
**Pull request by zzaakiirr**
### What does this PR do?

This PR improves `Command::Base#step` method

## Summary by CodeRabbit

- **New Features**
	- Enhanced retry mechanism in the command execution process, allowing users to define the maximum number of retries and wait time between attempts.
	- Added a method to suppress console output during command execution, improving control over logging and user interface cleanliness.

- **Bug Fixes**
	- Improved test coverage for the retry logic, ensuring reliable behavior in various scenarios.

- **Documentation**
	- Updated configuration for RSpec tests to enforce better organization and readability by limiting nested groups.

### CODE REVIEW

**Comment by zzaakiirr** [comment]
@coderabbitai review

File: lib/command/base.rb
Comment (zzaakiirr) [reply lib/command/base.rb]: Fixed + added test

File: lib/command/base.rb
Comment (zzaakiirr) [reply lib/command/base.rb]: Moved `run_count` assignment below to `if` block

Comment (zzaakiirr) [reply lib/command/base.rb]: There is no need to initialize `run_count` variable if `retry_on_failure` is `false`

**Review by rafaelgomesxyz — COMMENTED** [review]
Hm, I'm not sure that we want to do this. There are some cases where `step` may fail, but we _do_ want it to keep retrying infinitely, because it will eventually succeed. I'll see if I can locate those cases and what can be done about them, otherwise this would be a breaking change.

**Comment by zzaakiirr** [comment]
> Hm, I'm not sure that we want to do this. There are some cases where `step` may fail, but we _do_ want it to keep retrying infinitely, because it will eventually succeed. I'll see if I can locate those cases and what can be done about them, otherwise this would be a breaking change.

@rafaelgomesxyz We can set `Float::INFINITY` as default value for `max_retry_count` option, WDYT?
```
def step(message, abort_on_error: true, retry_on_failure: false, max_retry_count: Float::INFINITY, wait: 1)
```

**Review by rafaelgomesxyz — CHANGES_REQUESTED** [review]
@zzaakiirr Setting `Float::INFINITY` as the default value makes it so that this PR doesn't really change anything.

Let's leave this on hold for a bit, I'll get back to it in a few days.

**Comment by zzaakiirr** [comment]
> Setting `Float::INFINITY` as the default value makes it so that this PR doesn't really change anything.

Yes, it doesn't change current behaviour, but method becomes more flexible - we already have [issue](https://github.com/shakacode/control-plane-flow/issues/157) where we can profit from configurable wait time & retry count - we shouldn't wait infinite time if domain update fails, it's better to fail with error, IMHO

**Review by rafaelgomesxyz — CHANGES_REQUESTED** [review]
Let's proceed with this.

I've merged some changes to `Command::Base#step` to `main`, but they shouldn't conflict with yours. Still, it's good to rebase.

File: spec/command/base_spec.rb
Comment (zzaakiirr) [reply spec/command/base_spec.rb]: Replaced `subject` with `let`

File: spec/command/base_spec.rb
Comment (rafaelgomesxyz) [inline spec/command/base_spec.rb]: Let's bring this spec back and adjust it for the new `max_retry_count` default.

File: spec/command/base_spec.rb
Comment (rafaelgomesxyz) [reply spec/command/base_spec.rb:31]: Because of `super()`, and also because we're shadowing `options`. This is much easier to read:

```
let(:options_with_max_retry_count) { options.merge(max_retry_count: 1) }
```

Or:

```
let(:options) { base_options.merge(max_retry_count: 1) }
let(:options) { common_options.merge(max_retry_count: 1) }
```

File: spec/command/base_spec.rb
Comment (rafaelgomesxyz) [inline spec/command/base_spec.rb:41]: We don't need to break this down and add this comment, just do `eq(2)`.

**Review by rafaelgomesxyz — CHANGES_REQUESTED** [review]

File: spec/command/base_spec.rb
Comment (zzaakiirr) [reply spec/command/base_spec.rb]: Removed `super()` approach

File: lib/command/base.rb
Comment (zzaakiirr) [reply lib/command/base.rb]: Done

File: spec/command/base_spec.rb
Comment (rafaelgomesxyz) [inline spec/command/base_spec.rb:39]: We don't need this comment. Or if you really want to keep it, we should bring back the comment we had for line 53, and also add a comment for line 28.

File: spec/command/base_spec.rb
Comment (rafaelgomesxyz) [inline spec/command/base_spec.rb:43]: We're still shadowing `options`. I'd still go for:

```
let(:options_with_max_retry_count) { options.merge(max_retry_count: 1) }
```

**Review by rafaelgomesxyz — CHANGES_REQUESTED** [review]

**Review by rafaelgomesxyz — APPROVED** [review]
Thanks!

Just do one final thing before merging please: update the changelog (follow the pattern we use) to mention that we fixed a potential infinite loop that could happen in certain commands.

**Pull request merged**
